specs/validator/0_beacon-chain-validator.md

```diff
@@ -0,0 +1,383 @@
+    * `vote.eth1_data.deposit_root` is the deposit root of the eth1.0 deposit contract at the block defined by `vote.eth1_data.block_hash`.
+* If `D` is empty:
+    * Let `block_hash` be the block hash of the `ETH1_FOLLOW_DISTANCE`th ancestory of the head of the canonical eth1.0 chain.
+    * Let `deposit_root` be the the deposit root of the eth1.0 deposit contract at the block defined by `block_hash`.
+* If `D` is nonempty:
+    * Let `best_vote` be the member of `D` that has the highest `vote.eth1_data.vote_count`, breaking ties by favoring block hashes with higher associated block height.
+    * Let `block_hash = best_vote.eth1_data.block_hash`.
+    * Let `deposit_root = best_vote.eth1_data.deposit_root`.
+* Set `block.eth1_data = Eth1Data(deposit_root=deposit_root, block_hash=block_hash)`.
+
+##### Signature
+
+Set `block.signature = signed_proposal_data` where `signed_proposal_data` is defined as:
+
+```python
+proposal_data = ProposalSignedData(
+    slot=slot,
+    shard=BEACON_CHAIN_SHARD_NUMBER,
+    block_root=hash_tree_root(block),  # where `block.sigature == EMPTY_SIGNATURE
+)
+proposal_root = hash_tree_root(proposal_data)
+
+signed_proposal_data = bls_sign(
+    privkey=validator.privkey,  # privkey store locally, not in state
+    message=proposal_root,
+    domain=get_domain(
+        state.fork_data,  # `state` is the resulting state of `block` transition
+        state.slot,
+        DOMAIN_PROPOSAL,
+    )
+)
+```
+
+#### Block body
+
+##### Proposer slashings
+
+Up to `MAX_PROPOSER_SLASHINGS` [`ProposerSlashing`](https://github.com/ethereum/eth2.0-specs/blob/master/specs/core/0_beacon-chain.md#proposerslashing) objects can be included in the `block`. The proposer slashings must satisfy the verification conditions found in [proposer slashings processing](https://github.com/ethereum/eth2.0-specs/blob/master/specs/core/0_beacon-chain.md#proposer-slashings-1). The validator receives a small "whistleblower" reward for each proposer slashing found and included.
+
+##### Casper slashings
+
+Up to `MAX_CASPER_SLASHINGS` [`CasperSlashing`](https://github.com/ethereum/eth2.0-specs/blob/master/specs/core/0_beacon-chain.md#casperslashing) objects can be included in the `block`. The casper slashings must satisfy the verification conditions found in [casper slashings processing](https://github.com/ethereum/eth2.0-specs/blob/master/specs/core/0_beacon-chain.md#casper-slashings-1). The validator receives a small "whistleblower" reward for each casper slashing found and included.
+
+##### Attestations
+
+Up to `MAX_ATTESTATIONS` aggregate attestations can be included in the `block`. The attestations added must satisfy the verification conditions found in [attestation processing](https://github.com/ethereum/eth2.0-specs/blob/master/specs/core/0_beacon-chain.md#attestations-1). To maximize profit, the validator should attempt to add aggregate attestations that include the most available that have not previously been added on chain.
```

Comment:
that include attestations from the largest number of validators whose signatures from the same epoch have not previously been added on chain?